_posts/2019-01-30-get-started-libaktualizr.adoc

```diff
@@ -0,0 +1,264 @@
+in the shared state associated with std::future. All exceptions are subclassed
+from AktualizrException class.
+
+There is a sample application that shows how libaktualizr API can be used.
+The application is located in `src/hmi_stub`.
+
+=== API Description
+
+==== General management, configuration and control flow
+[source,cpp]
+Aktualizr::Aktualizr(boost::filesystem::path config)
+
+Constructs an Aktualizr instance based on the provided config.
+A config should at least contain information about provisioning credentials (see
+`docs/automatic-provisioning.adoc`, `docs/implicit-provisioning.adoc` or
+`hsm-provisioning.adoc` depending on the used provisioning type),
+and about the local storage which will be used to store updates and metadata.
+For description of all configuration options, please, refer to the
+`docs/configuration.adoc` document and to the `config` folder for configuration
+examples.
+
+[source,cpp]
+void Aktualizr::AddSecondary(const std::shared_ptr<Uptane::SecondaryInterface> &secondary)
+
+Add new secondary to aktualizr. Must be called before Initialize.
+
+[source,cpp]
+void Aktualizr::Initialize()
+
+Initialize aktualizr. Any secondaries should be added before making this
+call. This will provision with the server if required. This must be called
+before using any other aktualizr functions except AddSecondary.
+
+[source,cpp]
+boost::signals2::connection Aktualizr::SetSignalHandler(std::function<void(shared_ptr<event::BaseEvent>)> &handler)
+
+Set a callback to receive event notifications.
+Returns a signal connection object, which can be disconnected if desired.
+The events are defined in the `src/libaktualizr/primary/events.h` header.
+
+[source,cpp]
+void Aktualizr::Pause()
+
+Requests the currently running command to pause and freezes the command queue.
+All commands that were scheduled after the currently executed command will wait
+in the command queue until Resume() is issued.
+Commands that are issued after Pause() will be put on a command queue,
+but not executed until Resume() is called.
+
+The Pause() function returns immediately, while pausing the running command
+still may be in progress.
+
+The function has no effect in case execution was already paused.
+
+[source,cpp]
+void Aktualizr::Resume()
+
+Resumes the execution of a previously paused command and all subsequent commands
+in the command queue.
+Returns immediately. The function has no effect if the execution was not paused.
+
+[source,cpp]
+void Aktualizr::Abort()
+
+Requests the currently running command to abort and flushes the command queue.
+The Abort() function will block until the command queue is empty and no command
+is in executed state.
+
+Can also be called on a previously paused instance, but doesn't change
+it's paused state.
+If a paused command was aborted, next time it will start the execution
+from the place it was paused.
+
+[source,cpp]
+Aktualizr::~Aktualizr(Config &config)
+
+Calls Abort() and destroys the aktualizr object.
```

Comment:
It's a good idea to point out that `Abort()` is called in the destructor but, presented like that, it looks like it's a method of the api that can be used manually. Seasoned C++ developers won't be fooled but maybe have a note to detail that it shouldn't be called manually in most usages.

Reply:
I think I'll move this text to the Abort() call description.